### ISSUE TITLE
Support JSONSchema for Thrift TText protocol

### ISSUE THREAD

**Issue by Dogacel** [line/armeria#4601]
## Motivation:

Improve JSON Schema support for `Thrift` `TText` protocol to ramp up `DocService`.

Parent PR:
- #4518 

## Requirements:

- [ ] `Typedef` support (currently they are only `object`).
- [ ] `map<..., Enum>` support (container types sometimes miss enum definitions)
- [ ] Numerated enum support
- [x] `Union` support
- [x] `binary` support

## Testing

Compared schemas for 
- `FooService`
- `HelloService`
- `OnewayHelloService`
- `SleepService`

We can compare using `HBase` and `Cassandra` but I got kinda lazy and my Thrift knowledge is very low.

Here is the current status of the schema as of 31.12.2022:
- https://gist.github.com/Dogacel/143b1185b54973a54e11b89c219ca6d3

**Comment by minwoox** [comment]
> Typedef support

I completely forgot about this. 😓 Let me handle it.

> Numerated enum support

Did you menat `TypedefedEnum`?

**Comment by Dogacel** [comment]
> Did you menat `TypedefedEnum`?

No, I thought you can give numbers as enums (e.g. 1, 2) but JSON schema only accepts names (string).

**Comment by minwoox** [comment]
Oops, I think I don't get it. 😅 

> you can give numbers as enums 

Do you mean specifying numbers as enums when sending a request?
But, couldn't we get the information of enums when creating the JSON schema regardless of the  Numerated enum?

**Comment by Dogacel** [comment]
> Oops, I think I don't get it. 😅
> 
> > you can give numbers as enums
> 
> Do you mean specifying numbers as enums when sending a request? But, couldn't we get the information of enums when creating the JSON schema regardless of the Numerated enum?

Yeah, we do have more information (possible enum values are autocompleted). I just wanted to fully support existing features but if you think nobody uses numbers as enums while sending thrift requests then it should be fine.

**Comment by minwoox** [comment]
Ah ah, so you mean we can create the JSON schema with the information but we cannot support auto-completion when sending a request using numbers? Is that correct? 😄 
Then, probably we can specify all the values we support? 🤔 
```
{
  "enum": ["red", "amber", "green", 1, 2, 3]
}
```

**Comment by Dogacel** [comment]
> Ah ah, so you mean we can create the JSON schema with the information but we cannot support auto-completion when sending a request using numbers? Is that correct? 😄 Then, probably we can specify all the values we support? 🤔
> 
> ```
> {
>   "enum": ["red", "amber", "green", 1, 2, 3]
> }
> ```

Updated ticket description.

**Comment by minwoox** [comment]
> Typedef support (currently they are only object).

This will be supported by https://github.com/line/armeria/pull/4628
